ddtrace/tracer/option_test.go

```diff
@@ -468,6 +485,15 @@ func TestTracerOptionsDefaults(t *testing.T) {
 		})
 
 		t.Run("env-port", func(t *testing.T) {
+			t.Setenv("DD_DOGSTATSD_PORT", "123")
+			tracer := newTracer(WithAgentTimeout(2))
+			defer tracer.Stop()
+			c := tracer.config
+			assert.Equal(t, "localhost:8125", c.dogstatsdAddr)
+			assert.Equal(t, "localhost:8125", globalconfig.DogstatsdAddr())
```

Comment:
I assume we expect localhost:8125 because that would be what's reported by the agent by default. But I imagine loadAgentFeatures isn't returning any data because we don't test with an agent here right? 

Maybe we can test this by setting `c.agent.StatsdPort` directly to 8125, and confirm that this value is used over the user-defined one

Reply:
@mtoffl01 I think that wouldn't test the code as it'll run in the wild. It's better to test with the zero value, as it means it's meaningful (falling back to 8125).